docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+| Assert | Verifies that the action of the method under test behaves as expected. |
+
+This pattern ensures that unit tests are readable, self-describing, and consistent.
+
+## Dependency injection and unit testing
+
+One of the motivations for adopting a loosely-coupled architecture is that it facilitates unit testing. One of the types registered with the dependency injection service is the `IAppEnvironmentService` interface. The following code example shows an outline of this class:
+
+```csharp
+public class OrderDetailViewModel : ViewModelBase
+{
+    private IOrderService _ordersService;
+
+    public OrderDetailViewModel(
+        IAppEnvironmentService appEnvironmentService,
+        IDialogService dialogService, INavigationService navigationService, ISettingsService settingsService)
+        : base(dialogService, navigationService, settingsService)
+    {
+        _appEnvironmentService = appEnvironmentService;
+        _settingsService = settingsService;
+    }
+}
+```
+
+The `OrderDetailViewModel` class has a dependency on the `IAppEnvironmentService` type, which the dependency injection container resolves when it instantiates an `OrderDetailViewModel` object. However, rather than create an `IAppEnvironmentService` object which utilizes real servers, devices and configurations to unit test the `OrderDetailViewModel` class, instead, replace the `IAppEnvironmentService` object with a mock for the purpose of the tests.
+
+This approach allows the `IAppEnvironmentService` object to be passed into the `OrderDetailViewModel` class at runtime, and in the interests of testability, it allows a mock class to be passed into the `OrderDetailViewModel` class at test time. The main advantage of this approach is that it enables unit tests to be executed without requiring unwieldy resources such as runtime platform features, web services, or databases.
+
+## Testing MVVM applications
+
+Testing models and view models from MVVM applications is identical to testing any other classes, and uses the same tools and techniques; this includes features such as unit testing and mocking. However, some patterns that are typical to model and view model classes can benefit from specific unit testing techniques.
+
+> [!TIP]
+> Test one thing with each unit test.
+
+Don't be tempted to make a unit test exercise more than one aspect of the unit's behavior. Doing so leads to tests that are difficult to read and update. It can also lead to confusion when interpreting a failure.
+
+The eShopOnContainers mobile app uses [xUnit](https://xunit.github.io/) to perform unit testing, which supports two different types of unit tests:
+
+|Testing Type | Description |
+|---------|---------|
+| Facts | Tests that are always true, which test invariant conditions |
+| Theories | Tests that are only true for a particular set of data. |
+
+The unit tests included with the eShopOnContainers mobile app are fact tests, so each unit test method is decorated with the `[Fact]` attribute.
```

Comment:
```suggestion
The unit tests included with the eShopOnContainers mobile app are fact tests, so each unit test method is decorated with the `Fact` attribute.
```

No need to annotate the attribute as it is declared in code, instead just call it `Fact`.